internal/kubernetes/utils.go

```diff
@@ -194,6 +200,81 @@ func mergeMetadata(new *metav1.ObjectMeta, old metav1.ObjectMeta) {
 	new.SetOwnerReferences(mergeOwnerReferences(new.OwnerReferences, old.OwnerReferences))
 }
 
+func EnqueueRequestForOwnerFuncs(log logr.Logger) handler.EventHandler {
+	return &handler.Funcs{
+		CreateFunc: func(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+			enqueueRequestsForOwners(log, evt.Object, q)
+		},
+		DeleteFunc: func(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+			enqueueRequestsForOwners(log, evt.Object, q)
+		},
+		GenericFunc: func(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+			enqueueRequestsForOwners(log, evt.Object, q)
+		},
+		UpdateFunc: func(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+			oldOwners := getOwnersFromObject(e.ObjectOld)
+			newOwners := getOwnersFromObject(e.ObjectNew)
+
+			for _, newOwner := range newOwners {
+				if ownerSliceContains(oldOwners, newOwner) && hasController(newOwners) {
```

Comment:
I guess we have the possibility that an object has no controller and we would not reconcile. Maybe trigger reconciliation for the first owner in this case.